addons/mail/static/tests/qunit_suite_tests/components/message_tests.js

```diff
@@ -699,6 +699,35 @@ QUnit.test('prevent attachment delete on non-authored message in channels', asyn
     );
 });
 
+QUnit.test('allow attachment image download on message', async function (assert) {
+    assert.expect(2);
+
+    const { createMessageComponent, messaging } = await start();
+    const message = messaging.models['Message'].create({
+        attachments: insertAndReplace({
+            filename: "BLAH.jpg",
+            id: 10,
+            name: "BLAH",
+            mimetype: 'image/jpeg',
+        }),
+        author: replace(messaging.currentPartner),
+        body: "<p>Test</p>",
+        id: 100,
+    });
+    await createMessageComponent(message);
+
+    assert.containsOnce(
+        document.body,
+        '.o_AttachmentImage',
+        "should have an attachment",
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_AttachmentImage_actionDownload',
+        "should have download attachment button"
+    );
+});
```

Comment:
We are currently changing tests so that:
- Most data are made `MockServer`-side, with `pyEnv` (instead of locally with `messaging.models[].create()`, because the server has autority on existence of message)
- Make test with a root component like `Discuss` (instead of creating a small component like `createMessageComponent`, because this can easily break due to business code relying on some context of message being viewed).

Reply:
@alexkuhn Changes Done!